credoai/lens/lens.py

```diff
@@ -60,14 +60,13 @@ def __init__(
         self.command_list: list = []
         self.logger = global_logger
         self.pipeline_results: list = []
-        self._validate()
         if self.assessment_data and self.assessment_data.sensitive_features is not None:
             self.sens_feat_names = list(self.assessment_data.sensitive_features)
         else:
             self.sens_feat_names = []
-        # If a list of steps is passed create the pipeline
-        if pipeline:
-            self._generate_pipeline(pipeline)
+        self._generate_pipeline(pipeline)
```

Comment:
This is running everytime now, the code inside `_generate_pipeline()` does not account for an empty pipeline.
In `_generate_pipeline()` we could for example add something like:
```
if not pipeline:
  return
```